src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationAttributeStore.cs

```diff
@@ -230,8 +232,62 @@ internal bool TryGetPropertyStoreItem(string propertyName, [NotNullWhen(true)] o
                     propertyStoreItems[property.Name] = item;
                 }
 
+                return CombineMetadataTypeAttributes(propertyStoreItems);
+            }
+
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2075:DynamicallyAccessedMembersAttribute",
+                       Justification = "The ctor is marked with DynamicallyAccessedMembers.")]
+            private Dictionary<string, PropertyStoreItem> CombineMetadataTypeAttributes(Dictionary<string, PropertyStoreItem> propertyStoreItems)
+            {
+                // check if metadata attribute is present on the type
+                var metadataAttr = _type.CustomAttributes.FirstOrDefault(attr => attr.AttributeType.Name == nameof(MetadataTypeAttribute));
```

Comment:
Can you refactor 

[https://github.com/dotnet/runtime/blob/51d6f177e14c21eb4511b3e5a1d6fb61d90da61d/src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/AssociatedMetadataTypeTypeDescriptor.cs#L129-L133](https://github.com/dotnet/runtime/blob/51d6f177e14c21eb4511b3e5a1d6fb61d90da61d/src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/AssociatedMetadataTypeTypeDescriptor.cs#L129-L133)

and reuse the same logic here? That way we are consistent in how we are reading this attribute.

Once you do that, the trim warnings will go away because the Type returned from that refactored method can be annotated with `DynamicallyAccessedMemberTypes.All`, which means you can reflect over the Properties on that type.

Reply:
Thanks. I refactored it (not pushed yet) but it's complaining about possible nullability. I'll merge master in and see if it's a transitory issue.

Reply:
Using `(MetadataTypeAttribute)Attribute.GetCustomAttribute` I was getting:
`Error	CS8600	Converting null literal or possible null value to non-nullable type.`. 

I noticed in file [AssociatedMetadataTypeTypeDescriptor.cs](https://github.com/dotnet/runtime/blob/51d6f177e14c21eb4511b3e5a1d6fb61d90da61d/src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/AssociatedMetadataTypeTypeDescriptor.cs#L129-L133) it has `#nullable disable` which is suppressing the error.

I switched to `(MetadataTypeAttribute)Attribute.GetCustomAttributes(_type, typeof(MetadataTypeAttribute)).First()`